scripts/mc_launch_rtp_daemon.py

```diff
@@ -0,0 +1,200 @@
+    RENAME_BAD_FILES = True
+    BAD_SUFFIX = ".METADATA_ERROR"
+    CONDA_ENV = "RTP"
+    WORKING_DIRECTORY = "/home/obs/rtp_makeflow"
+    WORKFLOW_CONFIG = (
+        "/home/obs/src/hera_pipelines/pipelines/h6c/rtp/v2/h6c_rtp_stage_1.toml"
+    )
+
+    db = mc.connect_to_mc_db(None)
+    r = redis.Redis(REDISHOST, decode_responses=True)
+
+    while True:
+        jds = r.hgetall(JD_KEY)
+        jds = [jd for jd, val in jds.items() if int(val) == JD_READY]
+        if len(jds) == 0:
+            print("No JDs ready")
+            time.sleep(10)
+            continue
+
+        # just process one jd at a time
+        jd_redis = jds[0]
+        # set status to fail in case something happens
+        r.hset(JD_KEY, jd_redis, JD_FAIL)
+        jd = int(jd_redis)
+        print("Processing JD", jd)
+        # clear out this jd from uploaded queue
+        uploaded = []
+        for cnt in range(r.llen(UPLOADED_KEY)):
+            f = r.rpop(UPLOADED_KEY)
+            if f.startswith(str(jd_redis)):
+                # grab files for this jd
+                uploaded.append(f)
+            else:
+                # otherwise return key to queue
+                r.lpush(UPLOADED_KEY, f)
+            uploaded.append(r.rpop(UPLOADED_KEY))
+        print(f"Found {len(uploaded)} files for JD {jd} in redis")
+        filelist = []
+        with db.sessionmaker() as session:
+            results = session.get_rtp_launch_record_by_jd(jd)
+            if len(results) == 0:
+                # this shouldn't happen if processing works correctly
+                warnings.warn(
+                    f"No RTP launch records found for JD {jd}. Flagging as failed."
+                )
+                continue
+            for result in results:
+                # build full filename
+                filename = os.path.join(result.prefix, result.filename)
+                filelist.append(filename)
+        filelist = sorted(filelist)
+        print(f"Found {len(filelist)} files for JD {jd} in M&C")
+        existing_files = [f for f in filelist if os.path.exists(f)]
+        if len(existing_files) == 0:
+            warnings.warn("None of these files are on disk. Flagging as failed.")
+            continue
+
+        # scan files if desired
+        obsids = []
+        if SCAN_FILES:
+            print("Scanning files")
+            from pyuvdata import UVData
+
+            bad_metadata_files = []
+            for filename in filelist:
+                try:
+                    uvd = UVData()
+                    uvd.read(filename, read_data=False)
+                except (KeyError, OSError, ValueError):
+                    bad_metadata_files.append(filename)
+                    filelist.remove(filename)
+                    if RENAME_BAD_FILES:
+                        os.rename(filename, filename + BAD_SUFFIX)
+                else:
+                    # if file is valid, add obsid to running list
+                    starttime = Time(
```

Comment:
It would probably be better to use the `_get_obsids` function here too, since we use it below in the case that we're not scanning files. These methods are effectively equivalent, but I think it's better to be self-consistent.

Reply:
The `_get_obsids` function takes a list of filenames and gets the time_array out of each file, so it's not quite a drop in replacement here. Instead, I broke out the conversion of the time_array to an obsid as a new function that is called both here and in `get_obsids`.